src/classes/BDI_DataImport_BATCH.cls

```diff
@@ -39,47 +39,184 @@
-        strSoql += ' WHERE Status__c <> \'' + label.bdiImported + '\' ';   
+        strSoql += ' WHERE Status__c <> \'' + label.bdiImported + '\' ';
+
+        if (batchId != null) {
+            strSoql += ' AND NPSP_Data_Import_Batch__c = \'' + batchId + '\'';
+        } else {
+            strSoql += ' AND NPSP_Data_Import_Batch__c = NULL';
+        }
         // this ensures consistency for our test code, but also should
         // help users figure out import problems by importing in a consistent order.
-        strSoql += ' ORDER BY Name ';                   
+        strSoql += ' ORDER BY Name ';
+
+        // now load our settings
+        loadSettings(batchId);
+
+        // and make sure settings are valid
+        validateSettings(diSettings);
+    }
+
+    /*******************************************************************************************************
+    * @description loads the data import settings to use for this batch.  these may be either from
+    * custom settings, or if a specific BatchId was specified, the settings specified for that batch.
+    * @param batchId optional batch to scope processing to.
+    * @return void
+    */
+    private void loadSettings(ID batchId) {
+
+        // load up the default or specific settings
+        if (batchId == null) {
+            diSettings = UTIL_CustomSettingsFacade.getDataImportSettings();
+        } else {
+            list<DataImportBatch__c> listDIBatch = BDI_DataImportBatch_SEL.selectByIds(new Set<Id>{batchId});
+            if (listDIBatch.size() == 1) {
+                DataImportBatch__c diBatch = listDIBatch[0];
+                diSettings = new Data_Import_Settings__c(
+                    Account_Custom_Unique_ID__c = diBatch.Account_Custom_Unique_ID__c,
+                    Batch_Size__c = diBatch.Batch_Process_Size__c,
+                    Contact_Custom_Unique_ID__c = diBatch.Contact_Custom_Unique_ID__c,
+                    Contact_Matching_Rule__c = diBatch.Contact_Matching_Rule__c,
+                    Donation_Matching_Behavior__c = diBatch.Donation_Matching_Behavior__c,
+                    Donation_Matching_Implementing_Class__c = diBatch.Donation_Matching_Implementing_Class__c,
+                    Donation_Matching_Rule__c = diBatch.Donation_Matching_Rule__c,
+                    Donation_Date_Range__c = diBatch.Donation_Date_Range__c
+                );
+            } else {
+                throw(new BDIException(string.format(Label.bdiContactCustomIdError, new string[]{batchId})));
+            }
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description validates the data import settings.  Used by UI as well as at runtime by the batch.
+    * throws an exception if there are any invalid settings.
+    * @return void
+    */
+    public static void validateSettings(Data_Import_Settings__c diSettings) {
```

Comment:
Yesterday we discussed basing the settings component on the DataImportBatch__c object because it's a superset of the custom setting. Do you want to update this method to be similar for consistency?